addons/website_sale_collect/static/src/js/website_sale_configurators.js

```diff
@@ -0,0 +1,16 @@
+import { WebsiteSale } from '@website_sale/js/website_sale';
+
+WebsiteSale.include({
+    /**
+     * When click and collect is activated allow adding a product in the cart via configurator.
+     *
+     * @override of `website_sale`
+     */
+    _getAdditionalDialogProps() {
+        const props = this._super(...arguments);
+        props.isClickAndCollectActive = Boolean(
+            this.el.querySelector('.o_click_and_collect_availability')
+        );
```

Comment:
Discard my previous comment, It's currently not possible to do it in 18.0 and I don't want to hold back all the nice features in this PR just because this one doesn't work. We'll improve that later :)